### PR TITLE
Ticket 3 get topics endpoint

### PR DESCRIPTION
**Pull request by JRGPower**
Ticket code review request

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                        expect(topic).toEqual(
+                            expect.objectContaining({
+                                slug: expect.any(String),
+                                description: expect.any(String)
+                            })
+                        );
+                    });
+                })
+        });
+        test('GET 200 - all topics should have correct structure', () => {
```
Comment (cjplatten) [inline __tests__/app.test.js]: Is this test doing anything additional to the previous test?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+describe('ENDPOINT TESTS', () => {
+    describe('GET /api/topics', () => {
+        test('GET 200 - should return all an array with all topics from db', () => {
+            return request(app)
+                .get('/api/topics')
+                .expect(200)
+                .then((res) => {
+                    expect(res.body.topics).toBeInstanceOf(Array)
+                    expect(res.body.topics.length).toBe(3)
```
Comment (cjplatten) [inline __tests__/app.test.js]: specifying a number here has the potential to make this test brittle if new catergories were ever added to the test data, I'd recommend testing for greater than 0.

File: controllers/topics.controllers.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { selectTopics } = require("../models/topics.models")
+
+exports.getTopics = (req, res, next) => {
+    selectTopics().then((topics) => {
+        res.status(200).send({ topics })
+    })
```
Comment (cjplatten) [inline controllers/topics.controllers.js:8]: What happens in the case of an error?

File: README.md [hunk 4]
```diff
@@ -1,57 +1,10 @@
...
-```
-git branch -D <local branch>
-```
-
-## Husky
-
-To ensure we are not commiting broken code this project makes use of git hooks. Git hooks are scripts triggered during certain events in the git lifecycle. Husky is a popular package which allows us to set up and maintain these scripts. This project makes use a _pre-commit hook_. When we attempt to commit our work, the script defined in the `pre-commit` file will run. If any of our tests fail than the commit will be aborted.
-
-The [Husky documentation](https://typicode.github.io/husky/#/) explains how to configure Husky for your own project as well as creating your own custom hooks.\_
+Each file must contain `PGDATABASE=<database_name_here>`, with the correct database name for the test data and dev data respectively (see `/db/setup.sql` for the database names). Ensure that these .env files are .gitignored.
```
Comment (cjplatten) [inline README.md:10]: Editing the README was a task from ticket 2 - remember to submit a pr for each ticket and not combine multiple tickets

**Review by cjplatten — COMMENTED** [review]
PLease see comments

File: __tests__/app.test.js (hunk 1 above)
Comment (JRGPower) [reply __tests__/app.test.js]: Nope, not at the moment, the tests were supposed to have separate criteria but I accidentally duplicated part of the test. 

Corrected in [2f4ee93](https://github.com/JRGPower/nc-news-project/pull/1/commits/2f4ee939978a16495093c0c3db8f2770836dc208)

Comment (JRGPower) [reply __tests__/app.test.js]: I don't know best to reference the changes sorry

File: __tests__/app.test.js (hunk 2 above)
Comment (JRGPower) [reply __tests__/app.test.js]: Fixed as per your recommendation

File: README.md (hunk 4 above)
Comment (JRGPower) [reply README.md:10]: I'll make sure not to going forwards. Is there a way to do this in retrospect now?

File: controllers/topics.controllers.js (hunk 3 above)
Comment (JRGPower) [reply controllers/topics.controllers.js:8]: Fixed in [c5bbb58](https://github.com/JRGPower/nc-news-project/pull/1/commits/c5bbb58e0f661cf008737bb11c39a3888d85e4b9)

File: controllers/topics.controllers.js [hunk 5]
```diff
@@ -10,4 +10,8 @@ exports.getTopics = (req, res, next) => {
```
Comment (JVal-NC) [inline controllers/topics.controllers.js]: Error handlers are not controllers, so should not be located in a controller file/folder.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -20,18 +20,10 @@ describe('ENDPOINT TESTS', () => {
...
-                                slug: expect.any(String),
-                                description: expect.any(String)
-                            })
-                        );
-                    });
+                    expect(res.body.topics.length).toBeGreaterThan(0)
                 })
         });
-        test('GET 200 - all topics should have correct structure', () => {
+        test('GET 200 - all topic objects within array should have correct structure', () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js]: This test would benefit from a res.body.topics.length>0 assertion to avoid a false positive test when length=0.

**Review by JVal-NC — COMMENTED** [review]
See comments for changes required before merging. Please do another pull request.

**Pull request merged**